lib/senders/index.js

```diff
@@ -24,12 +23,49 @@ module.exports = function (config, log) {
       sms: config.sms
     }
   )
-  .then(
-    function (senders) {
-      var mailer = senders.email
-      mailer.sendVerifyCode = function (account, code, opts) {
-        return P.resolve(mailer.verifyEmail(
-          {
+  .then(senders => {
+    const ungatedMailer = senders.email
+    const configBounces = config.bounces || {}
+    const MAX_HARD = configBounces.hard && configBounces.hard.max || 0
+    const MAX_COMPLAINT = configBounces.complaint && configBounces.complaint.max || 0
+    const BOUNCE_TYPE_HARD = 1
+    const BOUNCE_TYPE_COMPLAINT = 3
+
+    function bounceGatedMailer(email) {
+      return db.emailBounces(email)
+        .then(bounces => {
+          let hards = 0
+          let complaints = 0
+          bounces.forEach(bounce => {
+            if (bounce.bounceType === BOUNCE_TYPE_HARD) {
+              hards++
+              if (hards >= MAX_HARD) {
```

Comment:
Doh, made this `> MAX`, so that does mean that a limit of 0 means that 1 bounce throws.